lib/jekyll/utils.rb

```diff
@@ -297,5 +287,36 @@ def merged_file_read_opts(site, opts)
       merged
     end
 
+    private
+    def merge_values(target, overwrite)
+      target.merge!(overwrite) do |_key, old_val, new_val|
+        if new_val.nil?
+          old_val
+        elsif mergable?(old_val) && mergable?(new_val)
+          deep_merge_hashes(old_val, new_val)
+        else
+          new_val
```

Comment:
We could probably get away with doing a `new_val.frozen? ? new_val.dup : new_val` here, or using a method like `unfreeze(new_val)` to do the same for both this and `old_val` on line 294.